src/pip/_internal/utils/misc.py

```diff
@@ -628,6 +641,12 @@ def unzip_file(filename, location, flatten=True):
                 fn = split_leading_dir(name)[1]
             fn = os.path.join(location, fn)
             dir = os.path.dirname(fn)
+            if not is_within_directory(location, fn):
+                raise InstallationError(
+                    'The zip file (%s) has a file (%s) trying to install '
+                    'outside target directory (%s)' %
```

Comment:
I would use the `format()` method on the string for new code instead of `%`.

Reply:
I second this suggestion, though this isn't a blocker for this PR.